.github/workflows/zip.yml

```diff
@@ -108,16 +139,70 @@ jobs:
            build-head \
            ${{ matrix.linking_type }}
     - uses: actions/upload-artifact@v4
-      if: ${{ always() }}
+      if: always()
       with:
         name: ${{ matrix.linking_type == 'static' && 'Firebase-actions-dir' || 'Firebase-actions-dir-dynamic' }}
         # Zip the entire output directory since the builder adds subdirectories we don't know the
         # name of.
         path: zip_output_dir
 
+  packaging_done:
+    runs-on: ubuntu-latest
+    needs: [package-head]
+    if: always()
+    outputs:
+      run_id: ${{ steps.get_run_id.outputs.run_id }}
+    steps:
+      - name: Check packaging result
+        if: ${{ needs.package-head.result == 'failure' }}
+        run: |
+          echo "Packaging failed. Aborting."
+          exit 1
+      - name: Get Run ID
+        id: get_run_id
+        run: |
+          if [[ -n "${{ github.event.inputs.zip_run_id }}" ]]; then
+            echo "run_id=${{ github.event.inputs.zip_run_id }}" >> $GITHUB_OUTPUT
+          elif [[ -n "${{ env.PINNED_RUN_ID }}" ]]; then
+            echo "run_id=${{ env.PINNED_RUN_ID }}" >> $GITHUB_OUTPUT
+          else
+            echo "run_id=${{ github.run_id }}" >> $GITHUB_OUTPUT
+          fi
+
+  check_framework_firestore_symbols:
```

Comment:
Note: This `check_framework_firestore_symbols` job was moved up to here with no other changes.  I wanted it so all jobs below were `quickstart_framework_*` jobs (further consolidation) coming in a following PR.